proposals/p1964.md

```diff
@@ -0,0 +1,356 @@
+type. However, explicitly converting this character literal to another character
+set might result in a character with a different value, but that still
+represents the newline character.
+
+```
+// 0x15 is the new line character in EBCDIC.
+Assert('\n' as EBCDICChar == EBCDICChar.Make(0x15));
+Assert('\u{A}' as EBCDICChar != EBCDICChar.Make(0x0A));
+```
+
+### Operations
+
+Character literals representing a single code point support the following
+operators:
+
+-   Comparison: `<`, `>`, `<=`, `>=` `==`
+-   Plus: `+`. However the behavior of this operator is different from its use
+    with strings. Rather than a concatenation, this will add the value of the
+    two characters:
+    -   If the `+` is used between a character literal representing a single
+        Unicode code point and an integer literal, this should produce a
+        character literal if the result fits in a Unicode code point.
+    -   If the `+` is used between two character literals this will produce an
+        error.
+-   Subtract: `-` Similar to the plus operator, this will subtract the value of
+    the two characters.
+
+    -   If the `-` is used between a character literal representing a single
+        code point or code unit and an integer literal, this should produce a
+        character literal as long as the result of the difference is in range.
+    -   If the `-` is used between two character literals, which are both a
+        single code point, this should produce an integer literal representing
+        the difference between the code points.
+
+    In cases where the `+` and `-` operators are used with a character literal
+    and a non-integer non-literal, the character literal should be converted to
+    the type of the other operand if possible. For example, in the expression
+    `w - 'a'`, where `w` is of type `Char8`, the `'a'` literal will be converted
+    to type `Char8`.
+
+There is intentionally no implicit conversion from character literals to integer
+types, but explicit conversions are permitted between character literals and
+integer types. Carbon will separate the integer types from character types
+entirely.
+
+## Rationale
+
+This proposal supports the goal of making Carbon code
+[easy to read, understand, and write](/docs/project/goals.md#code-that-is-easy-to-read-understand-and-write).
+Adding support for a specific character literal supports clean, readable,
+concise use and is a much more familiar concept that will make it easier to
+adopt Carbon coming from other languages. Have a distinct character literal will
+also allow us support useful operations designed to manipulate the literal's
+value. When working with `String`, we use the `+` operator to concatenate
+multiple `String`s, but say we wanted to advance a character to the next
+literal. Using a `String` will produce a type error, as we are misusing the `+`
+operator: `"a" + 1`. However with a character literal, we can support operations
```

Comment:
FWIW, I don't think it is decided that we will have the C++ `+` operator for concatenation on Carbon's String type -- I personally hope we don't.